llvm/lib/SYCLLowerIR/ESIMD/LowerESIMD.cpp

```diff
@@ -1834,6 +1834,15 @@ bool SYCLLowerESIMDPass::prepareForAlwaysInliner(Module &M) {
       continue;
     }
 
+    // If we are splitting by ESIMD, we are guarenteed the entire
```

Comment:
Is this code still needed after adding the special pass that remove those 2 attributes?

Reply:
Yes, I tested many combinations of with/without different changes in this PR and everything we currently have is required.